componentsPage.html

```diff
@@ -622,7 +622,33 @@ <h2>Example of Inline Lists</h2>
            </section>
 
            <section class="section" id="modal-container">
-            modal Will Come Here...
+            <h1>Modal</h1>
+            <p>Modals are pop-ups that are used to show important information to the user. <br>
+            Modals are different from Dialog Box as the user can't continue work on site until user responds to the Modal. <br>
+            While in case of Dialog Box, user can continue working on site without necessarily interacting with it.</p>
+            <br>
+            <br>
+            <h2>Modal Example</h2>
+            <div class="modalContainer">
+                <button onclick="showModal()" class="btn primary-btn">Show Modal</button>
+            </div>
+
+            <div class="modal-content">
+                <div class="modal">
+                    <button onclick="closeModal()" class="close-btn dismiss-btn">X</button>
```

Comment:
It's better to not keep JS in HTML, you can replace it by adding eventListener to buttons in JS file.